docs/tutorials/hpo-workflows/hpo-workflows.rst

```diff
@@ -0,0 +1,334 @@
+                    init_args: 
+                    train_proportion: 0.9
+                    validation_proportion: 0.1
+                - class_path: trainer.MyRayTrainer
+                    init_args:
+                    strategy: ddp
+                    epochs: 20
+                    checkpoints_location: checkpoints
+                    logger:
+                        class_path: itwinai.loggers.LoggersCollection
+                        init_args:
+                        loggers:
+                            - class_path: itwinai.loggers.MLFlowLogger
+                            init_args:
+                                experiment_name: MNIST Experiment
+                                log_freq: batch
+
+
+Okay, let's break down the arguments to our ``MyRayTorchTrainer`` class. 
+
+*   The ``scaling_config`` argument defines how we distribute resources between our trials. To learn more about the options for setting resources, please refer to the `ray train documentation <https://docs.ray.io/en/latest/train/user-guides/using-gpus.html>`_ on this topic. It is important that you ensure that you have allocated suffiecient resources on your cluster to be able to execute at least one trial. This means that if your configuration demands 4 GPUs and 32 CPUs per trial under ``resources_per_worker``, you should make sure that you have allocated at least this many GPUs and CPUs for your job.
+*   In the ``train_loop_config`` we define which hyperparameters we want to tune, as well as any additional parameters that we want to pass to our ``train()`` function. For the tunable parameters we have to specify the type and define their domain. For more information on which parameter types are possible and how to define their domains, have a look at `this page <https://docs.ray.io/en/latest/tune/api/search_space.html>`_, and learn how to define their domains according to the ``RayTorchTrainer``'s specifications `here <https://github.com/interTwin-eu/itwinai/blob/main/src/itwinai/torch/trainer.py>`_.
+*   In the ``tune_config`` we configure which search algorithm and scheduler to use to search the hyperparameter space and sample new configurations. Almost all search algorithms and schedulers supported by ray tune are also supported by us. You can refer to the ray documentation to learn more about the supported `search algorithms <https://docs.ray.io/en/latest/tune/api/suggestion.html#tune-search-al_>`_ and `schedulers <https://docs.ray.io/en/latest/tune/api/schedulers.html>`_. In the ``num_samples`` argument you can specify how many trials you wish to run, the default is one. Ray will queue trials if they cannot all be executed at once.
+*   The ``run_config`` defines a path that is used for checkpointing. This is mandatory to set if you want to distribute any one trial across more than one node, because ray uses this as a shared directory to coordinate and share data generated on each of the nodes.
```

Comment:
These lines are suddenly a lot longer than all other lines. Not that important from the perspective of the user reading the documentation, but might make it a bit worse if someone has to update it in the future.

Reply:
Yeah if I added line breaks it messed with the formatting when building the docs..🥲